subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -494,4 +496,26 @@ static int backend_init(const struct device *instance)
 			 CONFIG_IPC_SERVICE_REG_BACKEND_PRIORITY,			\
 			 &backend_ops);
 
-DT_INST_FOREACH_STATUS_OKAY(DEFINE_BACKEND_DEVICE)
+DT_INST_FOREACH_STATUS_OKAY(BACKEND_DEVICE_DEFINE)
+
+#define BACKEND_CONFIG_DEFINE(i) BACKEND_CONFIG_POPULATE(i),
+
+static int shared_memory_prepare(const struct device *arg)
+{
+	const struct backend_config_t *backend_config;
+	const struct backend_config_t backend_configs[] = {
+		DT_INST_FOREACH_STATUS_OKAY(BACKEND_CONFIG_DEFINE)
+	};
+
+	for (backend_config = backend_configs;
+	     backend_config < backend_configs + ARRAY_SIZE(backend_configs);
+	     backend_config++) {
+		if (backend_config->role == ROLE_HOST) {
+			memset((void *) backend_config->shm_addr, 0, VDEV_STATUS_SIZE);
+		}
+	}
```

Comment:
I think we could pull this off with a way smaller change. Maybe something like:
```
#define BACKEND_CONFIG_INIT(n) config[n] = &backend_config_##n;
static int shared_memory_prepare(const struct device *arg)
{
       static struct backend_config_t
               *config[DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)];

       DT_INST_FOREACH_STATUS_OKAY(BACKEND_CONFIG_INIT);

       for (int i = 0; DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT); i++) {
               ...
       }
}
SYS_INIT(shared_memory_prepare, PRE_KERNEL_1, 1);
```
without resorting to `BACKEND_CONFIG_POPULATE`.

Reply:
@kapi-no can you try using this solution? I guess the changed LOC will be smaller.

Reply:
@carlocaione, sure, I can change it anytime